File: src/main/java/me/rsman/BetterMinecraftCore/Enchantments/CustomEnchantClass.java
package me.rsman.BetterMinecraftCore.Enchantments;

import org.bukkit.NamespacedKey;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemStack;

import java.util.HashMap;
import java.util.Map;

public abstract class CustomEnchantClass extends Enchantment {
    public Map<String, Long> attributeModifiers;    public CustomEnchantClass(NamespacedKey key) {
        super(key);
    }
    public abstract boolean isApplicable(ItemStack item);
    public abstract int getMinimumLevel();
    public abstract boolean hasAttributesModifiers();
    public abstract Map<String, Long> getAttributesModifiers();
}

File: src/main/kotlin/me/rsman/BetterMinecraftCore/Enchantments/CustomEnchantClass.kt
package me.rsman.BetterMinecraftCore.Enchantments

import me.rsman.BetterMinecraftCore.enums.EAttributes
import org.bukkit.enchantments.Enchantment
import org.bukkit.NamespacedKey
import org.bukkit.inventory.ItemStack

abstract class CustomEnchantClass(key: NamespacedKey?) : Enchantment(key!!) {
    abstract val attributeModifiers: Map<EAttributes, Long>
    abstract fun isApplicable(item: ItemStack): Boolean
    abstract val minimumLevel: Int
    abstract fun hasAttributesModifiers(): Boolean
}